src/app/Plans/components/PlanDetails.tsx

```diff
@@ -0,0 +1,183 @@
+import * as React from 'react';
+import { Grid, GridItem, Popover, Button, List, Text, ListItem } from '@patternfly/react-core';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import { StatusIcon } from '@konveyor/lib-ui';
+
+import MappingDetailView from '@app/Mappings/components/MappingDetailView';
+import { IMetaObjectMeta, IPlan, IVMwareVM, MappingType, POD_NETWORK } from '@app/queries/types';
+import { useInventoryProvidersQuery, useMappingsQuery, useVMwareVMsQuery } from '@app/queries';
+import { usePausedPollingEffect } from '@app/common/context';
+import MappingStatus from '@app/Mappings/components/MappingStatus';
+import { warmCriticalConcerns, someVMHasConcern } from './Wizard/helpers';
+interface IPlanDetailsProps {
+  plan: IPlan;
+}
+
+const PlanDetails: React.FunctionComponent<IPlanDetailsProps> = ({ plan }: IPlanDetailsProps) => {
+  usePausedPollingEffect();
+
+  const networkMappings = useMappingsQuery(MappingType.Network);
+  const networkMapping =
+    networkMappings.data?.items.find(
+      (mapping) => (mapping.metadata as IMetaObjectMeta).name === plan.spec.map.network.name
+    ) || null;
+
+  const storageMappings = useMappingsQuery(MappingType.Storage);
+  const storageMapping =
+    storageMappings.data?.items.find(
+      (mapping) => (mapping.metadata as IMetaObjectMeta).name === plan.spec.map.storage.name
+    ) || null;
+
+  const providers = useInventoryProvidersQuery();
+  const provider =
+    providers.data?.vsphere.find((provider) => provider.name === plan.spec.provider.source.name) ||
+    null;
+
+  const vms = useVMwareVMsQuery(provider);
+  const selectedVMs =
+    vms.data?.filter((vm) => plan.spec.vms.find((planVM) => planVM.id === vm.id)) || [];
+
+  const warmCriticalConcernsFound = plan.spec.warm
+    ? warmCriticalConcerns.filter((label) => someVMHasConcern(selectedVMs as IVMwareVM[], label))
+    : [];
+
+  return (
+    <Grid hasGutter className={`${spacing.mtSm} ${spacing.mbMd}`}>
```

Comment:
There's some weirdness going on with the loading spinners here. As implemented, the first time I open a plan details modal, the mapping fields are blank for a moment, then they have spinners and then they display. I tried wrapping this whole thing with `<ResolvedQueries>` so it doesn't show anything until all 4 queries in this component are resolved, but then it doesn't start loading mapping resources until the mapping details are mounted anyway. I get spinners for the network mapping and the mapping statuses, but the storage mapping shows with missing resources (it's not waiting for the resource queries before it shows things).

I managed to fix it by forcing ALL the necessary queries to load up front, including the mappingResourceQueries for both network and storage mappings (which are used by MappingDetailView). It has one big spinner until everything is ready. I don't love this though, It shouldn't have been showing incomplete stuff at all, we may want to figure out how we can have the inner ResolvedQueries stuff work as intended. But this is a nice experience anyway.

This is the change, feel free to cherry-pick: https://github.com/konveyor/forklift-ui/commit/1132f67eb627be785209745285820cf2cd6aec7a

Reply:
Thanks for that, you're the query loader expert (among other things)!